lib/chkname.c

```diff
@@ -32,8 +32,8 @@ static bool is_valid_name (const char *name)
 	}
 
 	/*
-         * User/group names must match gnu e-regex:
-         *    [a-zA-Z0-9_.][a-zA-Z0-9_.-]{0,30}[a-zA-Z0-9_.$-]?
+         * User/group names must match BRE regex:
```

Comment:
i'd be inclined to link to `https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html` for the definition of BRE.  googling "BRE regex" turns up docs, but POSIX is like the 7th link down the page.

Reply:
I've added it to the commit message.

```
$ git range-diff shadow/master gh/validname validname 
1:  8ac64c14 ! 1:  22d753cb lib/chkname.c: Update regex for valid names
    @@ Commit message
         that has no length limit.
     
         Closes: <https://github.com/shadow-maint/shadow/issues/836>
    +    Link: <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html>
         Cc: Iker Pedrosa <ipedrosa@redhat.com>
    +    Cc: Mike Frysinger <vapier@gentoo.org>
         Signed-off-by: Alejandro Colomar <alx@kernel.org>
     
      ## lib/chkname.c ##
```

(I was even tempted to remove the comment from the source code and just put it in a commit message, to avoid it becoming stale again.)

Does that look good to you?